pyrobosim/pyrobosim/core/yaml.py

```diff
@@ -175,39 +175,44 @@ def add_global_path_planner(self):
         ):
             return
+
         planner_data = self.data["global_path_planner"]
         planner_type = planner_data["type"]
+        create_planner = False
 
         if planner_type == "search_graph":
-            max_edge_dist = planner_data.get("max_edge_dist", np.inf)
-            collision_check_dist = planner_data.get("collision_check_dist", 0.1)
-            self.world.create_search_graph(
-                max_edge_dist=max_edge_dist,
-                collision_check_dist=collision_check_dist,
-                create_planner=True,
-            )
-        else:
-            # Always make a search graph as we use it for other things.
-            max_edge_dist = planner_data.get("max_edge_dist", np.inf)
-            collision_check_dist = planner_data.get("collision_check_dist", 0.1)
-            self.world.create_search_graph(
-                max_edge_dist=max_edge_dist,
-                collision_check_dist=collision_check_dist,
-                create_planner=False,
+            create_planner = True
+
+        elif planner_type == "prm":
+            from pyrobosim.navigation.prm import PRMPlanner
+
+            max_nodes = planner_data.get("max_nodes", 100)
+            max_connection_dist = planner_data.get("max_connection_dist", 1.0)
+            self.world.path_planner = PRMPlanner(
+                self.world,
+                max_nodes=max_nodes,
+                max_connection_dist=max_connection_dist,
             )
 
-            if planner_type == "prm":
-                from pyrobosim.navigation.prm import PRMPlanner
+        elif planner_type == "astar_grid":
+            from pyrobosim.navigation.astar_grid import AStarGridPlanner
 
-                max_nodes = planner_data.get("max_nodes", 100)
-                max_connection_dist = planner_data.get("max_connection_dist", 1.0)
-                self.world.path_planner = PRMPlanner(
-                    self.world,
-                    max_nodes=max_nodes,
-                    max_connection_dist=max_connection_dist,
-                )
-            else:
-                warnings.warn(f"Invalid global planner type specified: {planner_type}")
+            resolution = planner_data.get("resolution", 0.05)
+            inflation_radius = planner_data.get("inflation_radius", 0.1)
+            self.world.path_planner = AStarGridPlanner(
+                self.world, resolution=resolution, inflation_radius=inflation_radius
+            )
+        else:
+            warnings.warn(f"Invalid global planner type specified: {planner_type}")
+
+        # Always make a search graph as we use it for other things.
+        max_edge_dist = planner_data.get("max_edge_dist", np.inf)
+        collision_check_dist = planner_data.get("collision_check_dist", 0.1)
+        self.world.create_search_graph(
+            max_edge_dist=max_edge_dist,
+            collision_check_dist=collision_check_dist,
+            create_planner=create_planner,
```

Comment:
Came across a weird bug while trying to navigate to a random goal in the demo.
The planner is the graph_planner.

This error came up when I pressed the navigate button.
```
 in show_planner_and_path
    self.axes.lines.remove(e)
AttributeError: 'ArtistList' object has no attribute 'remove'

```

Reply:
I ran into this with all the planners recently, but thought it was my environment being weird. In docker it worked.

I suspect there was a matplotlib change that may have broken things. Previously `self.axes.lines` returned a list of `Line` objects, but now it's an `ArtistList`. I'll have to dig into that first, I think

Reply:
Found the underlying bug : https://github.com/matplotlib/matplotlib/issues/22093

Reply:
Should be fixed by https://github.com/sea-bass/pyrobosim/pull/74

Reply:
>I just found an issue with this refactor, in that even if you set self.world.path_planner earlier, then it gets overwritten with a search graph here. So our demos always use search graph even if the world YAML sets PRM or A*.

when using the demo with `python pyrobosim/examples/demo.py --world-file test_world.yaml` with planner set to A*
The yaml loader is properly loading the A* planner .
I also dont see how it could get overwritten by
```
        # Optionally, create a global planner from the search graph
        if create_planner:
            self.path_planner = SearchGraphPlanner(self.search_graph)
```
Since this only creates a SearchGraphPlanner when `create_planner` is passed in as true.

Reply:
perhaps the `test_world.yaml` being used when you run is not the one you edited but the eariler one from the pyrobosim site-packages ?

Reply:
> perhaps the test_world.yaml being used when you run is not the one you edited but the eariler one from the pyrobosim site-packages ?

yup this is probably it. 
It should work as expected if you reinstall pyrobosim

Reply:
That might have been it. I'll test this branch again to make sure -- thanks!

Reply:
Worked for me! We can keep as it.